### Домашнее задание №6 ###

**Взаимосвязь величин. Параметрические и непараметрические показатели корреляции. Корреляционный анализ.**

**1. Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):<br>
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110];<br>
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].<br>
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy. Полученные значения должны быть равны.<br>
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков, а затем с использованием функций из библиотек numpy и pandas.**

In [2]:
import numpy as np
import pandas as pd

zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110]
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832]


sum_salary = 0
for item in zp:
    sum_salary+= item


sum_cs = 0
for item in ks:
    sum_cs+= item
    

sum_zp_cs=0
for ind in range(len(zp)):
    sum_zp_cs+=zp[ind]*ks[ind]

salary_mean = sum_salary/len(zp)
credscore_mean = sum_cs/len(ks)
salary_credscore_mean = sum_zp_cs/len(zp)

cov_math = salary_credscore_mean - salary_mean*credscore_mean
print(f"Ковариация с помощью элементарных действий: {cov_math}")

cov_numpy = np.cov(np.array(zp), np.array(ks), ddof=0)
print(f'Ковариация с помощью библиотеки numpy: {cov_numpy[1][0]}')



Ковариация с помощью элементарных действий: 9157.839999999997
Ковариация с помощью библиотеки numpy: 9157.84


In [3]:
sum_salary = 0
for item in zp:
    sum_salary+= (item-salary_mean)**2

sum_cs = 0
for item in ks:
    sum_cs+= (item-credscore_mean)**2
    
sigma_salary = (sum_salary/len(zp))**0.5
sigma_credscore = (sum_cs/len(ks))**0.5

corr_coeff = cov_math/(sigma_salary*sigma_credscore)

print(f"Коэффициент корреляции Пирсона с помощью элементарных действий: {corr_coeff}")

corr_coeff_np = np.corrcoef(np.array(zp), np.array(ks))

print(f"Коэффициент корреляции Пирсона с помощью библиотеки numpy: {corr_coeff_np[1][0]}")

Коэффициент корреляции Пирсона с помощью элементарных действий: 0.8874900920739158
Коэффициент корреляции Пирсона с помощью библиотеки numpy: 0.887490092073916


***Ответ: расхождения между значениями незначительные***

**2. Измерены значения IQ выборки студентов, обучающихся в местных технических вузах: 131, 125, 115, 122, 131, 115, 107, 99, 125, 111. Известно, что в генеральной совокупности IQ распределен нормально. Найдите доверительный интервал для математического ожидания с надежностью 0.95**

Из условия задачи:<br>
$n = 10$<br>
$\alpha = 1 - p = 1 - 0,95 = 0,05 $


Расчитаем среднее арифметическое для выборки $\overline{X}$:

In [4]:
X = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])
x=X.mean()
x

118.1

Расчитаем среднее квадратичное отклонение по выборке (несмещенное) $s_0$:

In [5]:
s0=np.std(X, ddof=1)
s0

10.54566788359614

Если неизвестно среднее квадратическое отклонение генеральной совокупности, то используем t-критерий: $T_{1,2} = \overline{X} \pm t_{\frac{\alpha}{2}} \cdot \frac{s_0}{\sqrt{n}}$<br>
Согласно t-таблице, для $\alpha = 0,025$ и числа степеней свободы равному $n-1 = 9$ значение $t = 2,262$

In [6]:
t1=x-2.262*s0/np.sqrt(10)
t2=x+2.262*s0/np.sqrt(10)
print(f'T1 = {t1}; T2 = {t2}')

T1 = 110.55660776308164; T2 = 125.64339223691834


***Ответ: $T_{1} = 110.55660776308164; T_{2} = 125.64339223691834$***

**3. Известно, что рост футболистов в сборной распределен нормально с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27, среднее выборочное составляет 174.2. Найдите доверительный интервал для математического ожидания с надежностью 0.95.**

Из условия задачи:<br>
$n = 27$<br>
$\sigma^2 = 25$<br>
$\mu = 174.2$<br>
$\alpha = 1 - p = 1 - 0,95 = 0,05 $


Так как среднекваlратичное у нас известно, $\sigma = \sqrt{\sigma^2} = \sqrt{25} = 5$ воспользуемся Z-критерием:<br>
### $Z_{\frac{\alpha}{2}} = Z_{0,025} = -1,96$ ###

Выполним расчет границ доверительного интервала по формуле:<br>
$T_{1,2} = \overline{X} \pm Z_{\frac{\alpha}{2}} \cdot \frac{\sigma}{\sqrt{n}}$

In [7]:
t1 = 174.2 - 1.96 * 5 /np.sqrt(27)
t2 = 174.2 + 1.96 * 5 /np.sqrt(27)
print(f'T1 = {t1}; T2 = {t2}')

T1 = 172.31398912064722; T2 = 176.08601087935276


***Ответ: $T_1 = 172.31398912064722; T_2 = 176.08601087935276$***